jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/ssl/X509Test.java

```diff
@@ -138,7 +154,7 @@ public void testClientClassWithSni() throws Exception
     {
         SslContextFactory clientSsl = new SslContextFactory.Client();
         Path keystorePath = MavenTestingUtils.getTestResourcePathFile("keystore_sni.p12");
-        clientSsl.setKeyStoreResource(Resource.newResource(keystorePath));
+        clientSsl.setKeyStoreResource(ResourceFactory.root().newResource(keystorePath));
```

Comment:
I'm starting to wonder about `.root()`.
What if we hid `.root()` (as private) and had instead a more clear method name on `ResourceFactory`?

``` java
Resource res = ResourceFactory.newUnmanagedResource(path);
```

Reply:
There isn't a single static method on `ResourceFactory` that returns `Resource` as that is reserved for `ResourceFactory` instances.

This proposal breaks that principle.

Reply:
It could be `Resource.newUnmanagedResource(path)` too, doesn't have to be `ResourceFactory`.

Internally, the implementation would still be ...

``` java
public static Resource newUnmanagedResource(path)
{
    return ResourceFactory.root().newResource(path);
}
```

That prevents users from using the returned `ResourceFactory` from the `.root()` call in other methods elsewhere that take a `ResourceFactory`.

Reply:
Only `ResourceFactory` instances can create `Resource` instances. Your proposal is breaking that other principle. :smiley: